test/LndClient/RPCSpec.hs

```diff
@@ -100,6 +104,13 @@ merchantEnv env = do
           }
     }
 
+btcClient :: IO Client
+btcClient = do
+  let user = btcRpcUser <$> btcEnv
+  let pass = btcRpcPassword <$> btcEnv
+  let url = btcRpcUrl <$> btcEnv
+  join (getClient <$> url <*> user <*> pass)
```

Comment:
I didn't meant this, but impressive how you are comfortable now with usage of functors and applicatives, that's cool) But there parsing still happens 3 times, I meant something like this (didn't checked it compiles, but you will get the idea):

```haskell
btcClient :: IO Client
btcClient = do
  env <- btcEnv
  let user = btcRpcUser btcEnv
  let pass = btcRpcPassword btcEnv
  let url = btcRpcUrl btcEnv
  getClient url user pass
```

Reply:
Aaah, make sense, anyway it will read it, sooner or later, but your code is much simpler, so changed as well)